### PR TITLE
refactor clay-space-edit to use Kiln 5

### PR DESCRIPTION
**Pull request by amycheng**
- updates UI logic to be compatible with Vue (which Kiln 4.0 uses)
- use Keen UI/material design (which Kiln 4.0 uses)
- fix drag and drop behavior
- adds readouts

### CODE REVIEW

**Comment by amycheng** [comment]
@jonwinton `clay-space-edit` is loaded before Kiln, so it doesn't have access to the ui components that it exposes.

If space-edit is loaded AFTER Kiln, then Vue is unable to register the custom selector buttons for the Edit UI and throws an error. This is because space-edit is not available yet when Kiln initializes.

File: src/ui/create-space-button.vue [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        return;
+      }
+
+      parentComponentName = window.kiln.utils.references.getComponentName(currentSelection.parentURI);
+      componentListName = currentSelection.parentField.path;
+
+      return get(this.$store.state.schemas, `[${parentComponentName}][${componentListName}]`);
+    },
+    /**
+     * Gran the available Spaces. Availability determined by the presence
```
Comment (reubenson) [inline src/ui/create-space-button.vue]: Gran?

File: src/ui/create-space-button.vue [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      const isSelected = !isUndefined(get(this.$store,'state.ui.currentSelection.uri'));
+      var ref;
+      if (isSelected) {
+        ref = this.$store.state.ui.currentSelection.uri;
+
+      // Check to make sure we:
+      // 1) Have Spaces available in the component list
+      // 2) we're not dealing with the `clay-space-edit` module
+      // 3) Check if we're dealing with a Space component
+        return this.availableSpaces.length && !checkIfSpaceEdit(ref) && !checkIfSpace(ref);
```
Comment (reubenson) [inline src/ui/create-space-button.vue]: !checkIfSpace(ref) is the opposite of (3) above?

**Comment by amycheng** [comment]
@jonwinton ACTUALLY, working on a [patch](https://github.com/clay/clay-kiln/pull/999) in Kiln that will fix this

**Comment by amycheng** [comment]
# Acceptance Criteria

[] Upgrade.js
[] User can create and edit spaces
[] Drag and drop
[] tests

File: circle.yml [hunk 3]
```diff
@@ -1,8 +0,0 @@
-machine:
```
Comment (jonwinton) [inline circle.yml]: Doesn't seem like Circle should be excluded. Are we not linting each PR or running tests? It looks like there's one test file in this PR and there are affordances for testing in the webpack config

File: package.json [hunk 4]
```diff
@@ -9,9 +9,8 @@
   },
   "scripts": {
     "lint": "eslint src",
-    "test": "npm run lint",
```
Comment (jonwinton) [inline package.json]: Doesn't feel like we should remove linting

Comment (amycheng) [reply package.json]: oops didn't see that was removed. I'll re-add it

File: circle.yml (hunk 3 above)
Comment (amycheng) [reply circle.yml]: k, i'll re-add Circle. I'll re-add testing when we actually have tests. The one test that Max wrote is completely outdated.

File: package.json (hunk 4 above)
Comment (amycheng) [reply package.json]: so it looks like the test command was removed but linting is still there. Added precommit linting

**Comment by amycheng** [comment]
@jonwinton could you 👀  this pr again? It passed [QA](https://trello.com/c/s3ZvhuHc/1616-users-should-be-able-to-edit-clay-spaces)

File: src/services/create-service.js [hunk 5]
```diff
@@ -1,256 +1,106 @@
...
-    references.select.select(el);
-    return el;
-  });
+    return spaceLogic;
+  } catch (err) {
+    throw new Error( 'The `content` field for a space must be a `_componentList` that contains exactly 1 Logic.'
+                     + ` Check the schema.yml for ${space}, it should be like this:\n`
+                     + 'content:\n  _componentList:\n    include:\n      space-logic-ads'
+                     + '\n ' + err.message);
+  }
```
[the remark this answers is not in the document]
Comment (amycheng) [reply src/services/create-service.js]: i dont think we can because we want to preserve the back ticks

File: src/ui/space-ui.vue [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+     * @param  {String} uri
+     * @return {Promise}
+     */
+    removeFromSpace(uri) {
+      const store = this.$store,
+        spaceRef = this.spaceRef;
+
+      removeLogic(this.$store, uri, this.items.length)
+        .then(function(){
+            setNewActive(store, spaceRef);
```
Comment (reubenson) [inline src/ui/space-ui.vue]: indentation is off

File: src/ui/space-ui.vue [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+     *
+     * @param  {String} iconName
+     * @return {String}
+     */
+    validateReadoutIcon(iconName) {
+      if (acceptedIcons.indexOf(iconName) > -1) {
+        return iconName;
+      } else {
+        console.warn(`Clay Space Logic: icon ${iconName} not found, no icon will be displayed`);
+
```
Comment (reubenson) [inline src/ui/space-ui.vue]: extra linebreak

File: src/ui/space-ui.vue [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+     * @param {Object} logicSchema with an optional _targeting field
+     * @return {Array<Object<String, String>>} {label, icon} for each target
+     *
+     */
+
+    createReadouts(logicData, logicSchema) {
+      // assumes all the readouts we'd want are set in the 'settings' group in
+      // the schema
+      const logicProps = get(logicSchema._groups.settings, 'fields'),
+      readoutProps = filter(logicProps, function (val) {
```
Comment (reubenson) [inline src/ui/space-ui.vue]: needs indentation

**Comment by amycheng** [comment]
@reubenson thanks for brining up a possible linter issue. so `eslint` was completely ignoring `.vue` files which why there was bad indentation and extra lines.

**Pull request merged**